test/mpi/datatype/gather_test.c

```diff
@@ -0,0 +1,53 @@
+#include <stdio.h>
+#include "mpi.h"
+#include "mpitest.h"
+
+#define BLKLEN (1024)
+#define NUM_BLKS (64)
+
+double large_buf[(BLKLEN + 1) * NUM_BLKS];
+double large_buf1[(BLKLEN + 1) * NUM_BLKS];
+
+int main(int argc, char *argv[])
+{
+    int errs = 0, nranks, rank, i, r = 0;
+
+    MTest_Init(&argc, &argv);
+
+    MPI_Comm_rank(MPI_COMM_WORLD, &rank);
+    MPI_Comm_size(MPI_COMM_WORLD, &nranks);
+
+    MPI_Request reqs[nranks-1];
+
+    MPI_Errhandler_set(MPI_COMM_WORLD, MPI_ERRORS_RETURN);
+
+    if (rank == 0) {
+        MPI_Datatype type;
+        MPI_Type_create_hvector(NUM_BLKS, BLKLEN, (BLKLEN + 1) * sizeof(double), MPI_DOUBLE, &type);
+        MPI_Type_commit(&type);
+
+        MPI_Irecv(large_buf, 1, type, r + 1, 111, MPI_COMM_WORLD, &reqs[r]);
+        MPI_Irecv(large_buf1, 1, type, r + 2, 111, MPI_COMM_WORLD, &reqs[r+1]);
+
+        MPI_Type_free(&type);
+
+        MPI_Waitall(nranks - 1, reqs, MPI_STATUSES_IGNORE);
+
+        for (i=0; i < (BLKLEN * NUM_BLKS); i++) {
+             if (large_buf[i] != rank + 1)
+		 ++errs;
+             if (large_buf1[i] != rank + 2)
+                 ++errs;
+        }
+    }
+    else {
+        for (i=0; i < (BLKLEN * NUM_BLKS); i++)
+             large_buf[i] = rank;
+             MPI_Send(large_buf, BLKLEN * NUM_BLKS, MPI_DOUBLE, 0, 111, MPI_COMM_WORLD);
```

Comment:
This indentation here is misleading. `MPI_Send` should be aligned with the `for` above.